clang/include/clang/AST/CanonBounds.h

```diff
@@ -50,7 +50,8 @@ namespace clang {
     enum class Result {
       LessThan,
       Equal,
-      GreaterThan
+      GreaterThan,
```

Comment:
I think this overloads Result in a confusing way.  Result is meant to represent the result of a lexicographic comparison, which produces total ordering.  It is confusing to now have NotEqual in Result.